libs/text-splitters/langchain_text_splitters/html.py

```diff
@@ -33,149 +34,443 @@ class ElementType(TypedDict):
     metadata: Dict[str, str]
 
 
+@dataclass
+class Node:
```

Comment:
Can we not rely on bs4 internal representation and avoid introducing new primitives in the implementation?